sympy/utilities/misc.py

```diff
@@ -355,7 +355,8 @@ def translate(s, a, b=None, c=None):
 
     mr = {}
     if a is None:
-        assert c is None
+        if c is not None:
+            raise TypeError('Called translate with 4 arguments and a second argument of None, expected 3 arguments if second argument is None')
```

Comment:
I'm not saying all of those are good practice, of course.  I'd be curious what SymPy's official policy is.